File: counting-elements/missing-integer.rb
# Given an array of integers, find the smallest positive integer that isn't included in the array

def solution(a)

  ints = Hash.new

  for i in 1..a.length + 1 do
    ints[i] = nil
  end

  a.each do |num|
    ints.delete(num)
  end

  ints.first[0]

end

File: counting-elements/missing_integer.py
# Given an array of integers, find the smallest positive integer that isn't included in the array

def solution(A):

    possInts = {}

    for i in range(1,len(A)+2):
        possInts[i] = None

    for i in A:
        possInts.pop(i, None)

    for key in possInts:
        return key

